File: 1894. Jogo de Varetas/VARETAS.java
import java.util.Scanner;

class VARETAS{
	public static void main(String[] args){
		int nComprimentos, nRetangulos;
		Scanner entrada = new Scanner(System.in);

		nComprimentos = entrada.nextInt();
		while(nComprimentos > 0){
			nRetangulos = 0;

			for(int i = 0; i < nComprimentos; i++){
				entrada.nextInt();
				nRetangulos += entrada.nextInt() / 2;
			}

			System.out.println(nRetangulos / 2);

			nComprimentos = entrada.nextInt();
		}
	}
}

File: 1894. Jogo de Varetas/varetas.c
#include <stdio.h>

int main(){
	int nComprimentos, nRetangulos, nVaretas;
	int i;

	scanf("%d", &nComprimentos);
	while(nComprimentos > 0){
		nRetangulos = 0;

		for(i = 0; i < nComprimentos; i++){
			scanf("%*d %d", &nVaretas);
			nRetangulos += nVaretas / 2;
		}
		printf("%d\n", nRetangulos / 2);

		scanf("%d", &nComprimentos);
	}

	return 0;
}
